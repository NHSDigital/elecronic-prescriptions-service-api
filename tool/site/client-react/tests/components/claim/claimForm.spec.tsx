import {staticProductInfo} from "./props"
import {render} from "@testing-library/react"
import pretty from "pretty"
import * as React from "react"
import ClaimForm from "../../../src/components/claim/claimForm"

test("Renders correctly", async () => {
  const component = <ClaimForm products={[staticProductInfo]} sendClaim={jest.fn}/>
  const {container} = render(component)
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

//TODO - test submit
//TODO - test reset
//TODO - test add endorsement
//TODO - test remove endorsement
