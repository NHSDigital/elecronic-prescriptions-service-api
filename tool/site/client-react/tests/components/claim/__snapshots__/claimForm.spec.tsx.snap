// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
"<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Methotrexate 10mg/0.2ml solution for injection pre-filled syringes</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">Dispensed</dd>
        </div>
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Quantity Dispensed</dt>
          <dd class=\\"nhsuk-summary-list__value\\">1 pre-filled disposable injection</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\">
        <div class=\\"nhsuk-checkboxes\\" id=\\"products.0.patientPaid.boxes\\">
          <div class=\\"nhsuk-checkboxes__item\\"><input class=\\"nhsuk-checkboxes__input\\" name=\\"products.0.patientPaid\\" id=\\"products.0.patientPaid.box\\" type=\\"checkbox\\" value=\\"false\\"><label class=\\"nhsuk-label nhsuk-checkboxes__label\\" id=\\"products.0.patientPaid.box--label\\" for=\\"products.0.patientPaid.box\\">Patient Paid</label></div>
        </div>
      </div>
      <div><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"button\\">Add Endorsement</button></div>
    </fieldset>
  </div>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Prescription Charge Exemption</legend>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"exemption.code--label\\" for=\\"exemption.code\\">Exemption Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"exemption.code--label\\" name=\\"exemption.code\\" id=\\"exemption.code\\">
          <option value=\\"0001\\">Patient has paid appropriate charges</option>
          <option value=\\"0002\\">is under 16 years of age</option>
          <option value=\\"0003\\">is 16, 17 or 18 and in full-time education</option>
          <option value=\\"0004\\">is 60 years of age or over</option>
          <option value=\\"0005\\">has a valid maternity exemption certificate</option>
          <option value=\\"0006\\">has a valid medical exemption certificate</option>
          <option value=\\"0007\\">has a valid prescription pre-payment certificate</option>
          <option value=\\"0008\\">has a War Pension exemption certificate</option>
          <option value=\\"0009\\">is named on a current HC2 charges certificate</option>
          <option value=\\"0010\\">was prescribed free-of-charge contraceptives</option>
          <option value=\\"0011\\">gets income support (IS)</option>
          <option value=\\"0012\\">gets income based Job Seeker's Allowance (JSA (IB))</option>
          <option value=\\"0013\\">is entitled to, or named on a VALID NHS tax credit exemption certificate</option>
          <option value=\\"0014\\">has a partner who gets Pension Credit Guarantee Credit (PGCC)</option>
          <option value=\\"0015\\">Patient does not need to pay the prescription charge</option>
        </select></div>
      <div class=\\"nhsuk-form-group\\">
        <div class=\\"nhsuk-checkboxes\\" id=\\"exemption.evidenceSeen.boxes\\">
          <div class=\\"nhsuk-checkboxes__item\\"><input class=\\"nhsuk-checkboxes__input\\" name=\\"exemption.evidenceSeen\\" id=\\"exemption.evidenceSeen.box\\" type=\\"checkbox\\" value=\\"false\\"><label class=\\"nhsuk-label nhsuk-checkboxes__label\\" id=\\"exemption.evidenceSeen.box--label\\" for=\\"exemption.evidenceSeen.box\\">Evidence Seen</label></div>
        </div>
      </div>
    </fieldset>
  </div><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Claim</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"reset\\">Reset</button>
</form>"
`;
